js/src/forum/components/NotificationList.js

```diff
@@ -97,27 +94,34 @@ export default class NotificationList extends Component {
     );
   }
 
-  config(isInitialized, context) {
-    if (isInitialized) return;
+  oncreate(vnode) {
+    super.oncreate(vnode);
+
+    const $notifications = this.$('.NotificationList-content');
+    const $scrollParent = $notifications.css('overflow') === 'auto' ? $notifications : $(window);
```

Comment:
I'm not sure I like explicitly tying our business logic here to the physical DOM. I'm also not sure if this is even allowed, seeing as the actual DOM can, at times, be regenerated to keep up with Mithril. This isn't a problem with SuperTextArea in Composer since that's reset on every single call to view(), but it might  be here. All for avoiding unnecesary repetition, but I don't think that's the way.

Reply:
It's the same concept you're applying for the `boundScrollHandler` - that one is an event handler, this one is an element.

I don't think this is business logic at all, it's DOM state management. What you attach manually in `oncreate`, you also need to detach in `onremove`.

> seeing as the actual DOM can, at times, be regenerated to keep up with Mithril.

For such a simple DOM structure, Mithril will re-use the DOM nodes, it only re-creates the VDOM repeatedly.

Another small problem with recalculating this in `onremove` is that you could theoretically end up with a different `$scrollParent`, which doesn't have the scroll handler. Since this is about removing a particular handler instance from a particular DOM node, IMO it can be stored. Anyway, too many words. :speak_no_evil:

Reply:
For some reason, I thought you had been talking about the `state` state. Like `NotificationListState`. Sorry, that's a misunderstanding on my part. Fixed.